This month in mathlib (Nov 2021) (#22)
leanprover-community/blog: posts/month-in-mathlib-nov-2021.md

---
author: 'Mathlib community'
category: 'month-in-mathlib'
date: 2021-12-01 06:58:04+00:00
description: ''
has_math: true
link: ''
slug: month-in-mathlib-nov-2021
tags: ''
title: This month in mathlib (Nov 2021)
type: text
---

This post summarizes some of the activity that happened in mathlib in November.

## Highlighted PRs

* Differential equations.
  - [PR #9791](https://github.com/leanprover-community/mathlib/pull/9791) Prove Picard-Lindelöf/Cauchy-Lipschitz Theorem

* Combinatorics.
  - [PR #10029](https://github.com/leanprover-community/mathlib/pull/10029) Hindman's theorem on finite sums

* Measure theory.
  - [PR #9920](https://github.com/leanprover-community/mathlib/pull/9920) Define uniform pmf on an inhabited fintype
  - [PR #10070](https://github.com/leanprover-community/mathlib/pull/10070) More on Radon-Nikodym derivatives
  - [PR #10072](https://github.com/leanprover-community/mathlib/pull/10072) An ae measurability criterion for ennreal-valued functions
  - [PR #10057](https://github.com/leanprover-community/mathlib/pull/10057) Define Vitali families

> 

Reviewer: ```suggestion
- [PR #10101](https://github.com/leanprover-community/mathlib/pull/10057) Differentiation of measures

```

---
author: 'Mathlib community'
category: 'month-in-mathlib'
date: 2021-12-01 06:58:04+00:00
description: ''
has_math: true
link: ''
slug: month-in-mathlib-nov-2021
tags: ''
title: This month in mathlib (Nov 2021)
type: text
---

This post summarizes some of the activity that happened in mathlib in November.

## Highlighted PRs

* Differential equations.
  - [PR #9791](https://github.com/leanprover-community/mathlib/pull/9791) Prove Picard-Lindelöf/Cauchy-Lipschitz Theorem

* Combinatorics.

>   - [PR #10029](https://github.com/leanprover-community/mathlib/pull/10029) Hindman's theorem on finite sums

Reviewer 2: The first one came a bit early and the last one a bit late, but we could mention the push of Kruskal-Katona towards mathlib:
```suggestion
  - [PR #10029](https://github.com/leanprover-community/mathlib/pull/10029) Hindman's theorem on finite sums
  - Set families are making their way to mathlib with [PR #9926](https://github.com/leanprover-community/mathlib/pull/9926), [PR #10223](https://github.com/leanprover-community/mathlib/pull/10223), [PR #10238](https://github.com/leanprover-community/mathlib/pull/10238) respectively defining antichains, shadow of a set family, UV compression. Those are ingredients for the LYM inequality, Sperner's theorem and the Kruskal-Katona theorem.
```
  -

* Measure theory.
  - [PR #9920](https://github.com/leanprover-community/mathlib/pull/9920) Define uniform pmf on an inhabited fintype
  - [PR #10070](https://github.com/leanprover-community/mathlib/pull/10070) More on Radon-Nikodym derivatives
  - [PR #10072](https://github.com/leanprover-community/mathlib/pull/10072) An ae measurability criterion for ennreal-valued functions
  - [PR #10057](https://github.com/leanprover-community/mathlib/pull/10057) Define Vitali families

> - [PR #10101](https://github.com/leanprover-community/mathlib/pull/10101) Differentiation of measures

Reviewer: ```suggestion
  - [PR #10101](https://github.com/leanprover-community/mathlib/pull/10101) Differentiation of measures
```

* Analysis.

>   - [PR #10145](https://github.com/leanprover-community/mathlib/pull/10145) Introduce `C*`-algebras
>   - [PR #10390](https://github.com/leanprover-community/mathlib/pull/10390) Define spectrum and prove basic prope...

Reviewer 3: And for this work on `C*`-algebras, it's early stages but #10530 is as interesting/more interesting than #10390 I think.
```suggestion
  - [PR #10145](https://github.com/leanprover-community/mathlib/pull/10145) Introduce `C*`-algebras
  - [PR #10390](https://github.com/leanprover-community/mathlib/pull/10390) Define spectrum and prove basic prope...
  - [PR #10530](https://github.com/leanprover-community/mathlib/pull/10530) Prove properties of spectrum in a Banach algebra
```

>   - [PR #9840](https://github.com/leanprover-community/mathlib/pull/9840) Rayleigh quotient produces eigenvectors

Reviewer 3: It's my own work, but I would like to mention the two PRs on diagonalization, part of the same project as the Rayleigh quotient PR:
```suggestion
  - [PR #9840](https://github.com/leanprover-community/mathlib/pull/9840) Rayleigh quotient produces eigenvectors
  - [PR #9995](https://github.com/leanprover-community/mathlib/pull/9995) diagonalization of self-adjoint endomorphisms (finite dimension)
  - [PR #10317](https://github.com/leanprover-community/mathlib/pull/10317) more concrete diagonalization theorem

* Combinatorics.
  - [PR #10029](https://github.com/leanprover-community/mathlib/pull/10029) Hindman's theorem on finite sums
  - Set families are making their way to mathlib with
  [PR #9926](https://github.com/leanprover-community/mathlib/pull/9926),
  [PR #10223](https://github.com/leanprover-community/mathlib/pull/10223),
  [PR #10238](https://github.com/leanprover-community/mathlib/pull/10238)
  respectively defining antichains, shadow of a set family, UV compression.
  Those are ingredients for the LYM inequality, Sperner's theorem and the Kruskal-Katona theorem.

* Measure theory.
  - [PR #9920](https://github.com/leanprover-community/mathlib/pull/9920) Define uniform pmf on an inhabited fintype
  - [PR #10070](https://github.com/leanprover-community/mathlib/pull/10070) More on Radon-Nikodym derivatives
  - [PR #10072](https://github.com/leanprover-community/mathlib/pull/10072) An ae measurability criterion for ennreal-valued functions
  - [PR #10057](https://github.com/leanprover-community/mathlib/pull/10057) Define Vitali families
  - [PR #10101](https://github.com/leanprover-community/mathlib/pull/10101) Differentiation of measures

> 

Reviewer: ```suggestion
This sequence of PRs shows the main theorem on differentiation of measures: given two measures `ρ` and `μ` on a finite-dimensional real vector space, the ratio `ρ (ball x r) / μ (ball x r)` converges `μ`-almost everywhere to the Radon-Nikodym derivative of `ρ` with respect to `μ`. The theorem is in fact proved in greater generality (Vitali families, as in Federer's famous book on geometric measure theory).
```

  - [PR #9995](https://github.com/leanprover-community/mathlib/pull/9995) diagonalization of self-adjoint endomorphisms (finite dimension)
  - [PR #10317](https://github.com/leanprover-community/mathlib/pull/10317) more concrete diagonalization theorem
  - [PR #10335](https://github.com/leanprover-community/mathlib/pull/10335) Continuous affine maps are smooth

* Convergence.
  - [PR #10073](https://github.com/leanprover-community/mathlib/pull/10073) Convergence of a sequence which does not oscillate infinitely
  - [PR #10258](https://github.com/leanprover-community/mathlib/pull/10258) Prove that, if u_n is subadditive, then u_n / n converges.

* Seminorms
  - [PR #9097](https://github.com/leanprover-community/mathlib/pull/9097) Define the Minkowski functional
  - [PR #10203](https://github.com/leanprover-community/mathlib/pull/10203) A seminorm is convex

* Homotopy theory.
  - [PR #9865](https://github.com/leanprover-community/mathlib/pull/9865) Add `homotopic` for `continuous_map`s.
  - [PR #10195](https://github.com/leanprover-community/mathlib/pull/10195) The functor from `Top` to `Groupoid`

* Number theory.

>   - [PR #9071](https://github.com/leanprover-community/mathlib/pull/9071) Define the class number of number fields and function fields

Reviewer 4: Let's link to the relevant blog post here:
```suggestion
  - [PR #9071](https://github.com/leanprover-community/mathlib/pull/9071) Define the class number of number fields and function fields
  This finishes the finiteness proof of the class group of a number field and function field, described in [a previous blog post](https://leanprover-community.github.io/blog/posts/dedekind-domains-and-class-number-in-lean/).
```

  - [PR #10203](https://github.com/leanprover-community/mathlib/pull/10203) A seminorm is convex

* Homotopy theory.
  - [PR #9865](https://github.com/leanprover-community/mathlib/pull/9865) Add `homotopic` for `continuous_map`s.
  - [PR #10195](https://github.com/leanprover-community/mathlib/pull/10195) The functor from `Top` to `Groupoid`

* Number theory.
  - [PR #9071](https://github.com/leanprover-community/mathlib/pull/9071) Define the class number of number fields and function fields
  This finishes the finiteness proof of the class group of a number field and function field, described in [a previous blog post](https://leanprover-community.github.io/blog/posts/dedekind-domains-and-class-number-in-lean/).
  - [PR #8820](https://github.com/leanprover-community/mathlib/pull/8820) Add theorem for Lucas primality test

* Group theory.
  - [PR #10059](https://github.com/leanprover-community/mathlib/pull/10059) (Infinite) Sylow subgroups are isomorphic
  - [PR #10283](https://github.com/leanprover-community/mathlib/pull/10283) Prove the full Schur-Zassenhaus theorem
  - [PR #10249](https://github.com/leanprover-community/mathlib/pull/10249) Define the exponent of a group

> * Sheafification.

Reviewer 5: It may be worthwhile to mention that these PRs work with with a general Grothendieck topology, as opposed to a topological space (where sheafification of type-valued presheaves has existed for some time now).

Author: Done

  An important building block in setting up for algebraic geometry,
  as well as a requirement for the Liquid Tensor Experiment.
  - [PR #10284](https://github.com/leanprover-community/mathlib/pull/10284) `P⁺` for a presheaf `P`.
  - [PR #10297](https://github.com/leanprover-community/mathlib/pull/10297) If `P` is a sheaf, then the map from `P` to `P⁺` is an isomorphism.
  - [PR #10303](https://github.com/leanprover-community/mathlib/pull/10303) The sheafification of a presheaf.
  - [PR #10328](https://github.com/leanprover-community/mathlib/pull/10328) `Sheaf_to_presheaf` creates limits
  - [PR #10334](https://github.com/leanprover-community/mathlib/pull/10334) `Sheaf J D` has colimits.
  - [PR #10401](https://github.com/leanprover-community/mathlib/pull/10401) `SheafedSpace` has colimits
  A side product is the important corollary that the category of sheaves

>   with values in a suitable concrete abelian category is itself an abelian category.

Reviewer 5: This is not completely accurate -- more work is required to obtain this result (e.g. the fact that sheafification is left exact, and a more general version of the fact that filtered colimits commute with finite limits) -- both of these have been done in LTE, but not yet in mathlib.

Author: Hmm, that's right. I will reword.

---
author: 'Mathlib community'
category: 'month-in-mathlib'
date: 2021-12-01 06:58:04+00:00
description: ''
has_math: true
link: ''
slug: month-in-mathlib-nov-2021
tags: ''
title: This month in mathlib (Nov 2021)
type: text
---

This post summarizes some of the activity that happened in mathlib in November.

## Highlighted PRs

* Differential equations.
  - [PR #9791](https://github.com/leanprover-community/mathlib/pull/9791) Prove Picard-Lindelöf/Cauchy-Lipschitz Theorem

* Combinatorics.
  - [PR #10029](https://github.com/leanprover-community/mathlib/pull/10029) Hindman's theorem on finite sums
  - Set families are making their way to mathlib with
  [PR #9926](https://github.com/leanprover-community/mathlib/pull/9926),
  [PR #10223](https://github.com/leanprover-community/mathlib/pull/10223),
  [PR #10238](https://github.com/leanprover-community/mathlib/pull/10238)
  respectively defining antichains, shadow of a set family, UV compression.

>   Those are ingredients for the LYM inequality, Sperner's theorem and the Kruskal-Katona theorem.

Reviewer 6: It would be nice to include wikipedia links for those esoteric combinatorics theorems (especially when hidden behind an acronym like LYM).

Reviewer 2: I'll take the esoteric 🤣 
```suggestion
  Those are ingredients for the [LYM inequality][https://en.wikipedia.org/wiki/Lubell-Yamamoto-Meshalkin_inequality], [Sperner's theorem][https://en.wikipedia.org/wiki/Sperner's_theorem] and the [Kruskal-Katona theorem][https://en.wikipedia.org/wiki/Kruskal-Katona_theorem].
```

* Measure theory.

>   - [PR #9920](https://github.com/leanprover-community/mathlib/pull/9920) Define uniform pmf on an inhabited fintype

Reviewer 6: We should at least replace pmf by its meaning.

Author: I ended up just removing this entry. It's actually not *that* spectacular mathematically.

  - [PR #10070](https://github.com/leanprover-community/mathlib/pull/10070) More on Radon-Nikodym derivatives
  - [PR #10072](https://github.com/leanprover-community/mathlib/pull/10072) An ae measurability criterion for ennreal-valued functions
  - [PR #10057](https://github.com/leanprover-community/mathlib/pull/10057) Define Vitali families
  - [PR #10101](https://github.com/leanprover-community/mathlib/pull/10101) Differentiation of measures

> This sequence of PRs shows the main theorem on differentiation of measures: given two measures `ρ` and `μ` on a finite-dimensional real vector space, the ratio `ρ (ball x r) / μ (ball x r)` converges `μ`-almost everywhere to the Radon-Nikodym derivative of `ρ` with respect to `μ` when `r` tends to `0`. The theorem is in fact proved in greater generality (Vitali families, as in Federer's famous book on geometric measure theory).

Reviewer 6: I don't think we need to list all PRs in this series. I would keep only the last two.

Author: Done

* Analysis.
  - [PR #10145](https://github.com/leanprover-community/mathlib/pull/10145) Introduce `C*`-algebras
  - [PR #10390](https://github.com/leanprover-community/mathlib/pull/10390) Define spectrum and prove basic prope...
  - [PR #10530](https://github.com/leanprover-community/mathlib/pull/10530) Prove properties of spectrum in a Banach algebra
  - [PR #9840](https://github.com/leanprover-community/mathlib/pull/9840) Rayleigh quotient produces eigenvectors
  - [PR #9995](https://github.com/leanprover-community/mathlib/pull/9995) diagonalization of self-adjoint endomorphisms (finite dimension)

>   - [PR #10317](https://github.com/leanprover-community/mathlib/pull/10317) more concrete diagonalization theorem

Reviewer 6: I don't understand why those diagonalization theorems are classified as analysis.

Reviewer 3: I put them there just because they depended on the Rayleigh quotient PR.  I'll separate all three to a separate category.

>   - [PR #10335](https://github.com/leanprover-community/mathlib/pull/10335) Continuous affine maps are smooth

Reviewer 6: I love that this was formalized as part of the sphere eversion project, but it's definitely not worth a mention here.

Reviewer 3: I deleted it (for now, someone else should feel free to bring it back)

> * Analysis.
>   - [PR #10145](https://github.com/leanprover-community/mathlib/pull/10145) Introduce `C*`-algebras
>   - [PR #10390](https://github.com/leanprover-community/mathlib/pull/10390) Define spectrum and prove basic prope...
>   - [PR #10530](https://github.com/leanprover-community/mathlib/pull/10530) Prove properties of spectrum in a Banach algebra
>   - [PR #9840](https://github.com/leanprover-community/mathlib/pull/9840) Rayleigh quotient produces eigenvectors
>   - [PR #9995](https://github.com/leanprover-community/mathlib/pull/9995) diagonalization of self-adjoint endomorphisms (finite dimension)
>   - [PR #10317](https://github.com/leanprover-community/mathlib/pull/10317) more concrete diagonalization theorem
>   - [PR #10335](https://github.com/leanprover-community/mathlib/pull/10335) Continuous affine maps are smooth

Reviewer 3: ```suggestion
* Hilbert spaces
  - [PR #9840](https://github.com/leanprover-community/mathlib/pull/9840) Rayleigh quotient
  - [PR #9995](https://github.com/leanprover-community/mathlib/pull/9995) diagonalization theorem
  - [PR #10317](https://github.com/leanprover-community/mathlib/pull/10317) more concrete diagonalization theorem
This sequence of PRs establishes the connection between eigenvalues and the Rayleigh quotient, and proves the diagonalization theorem for self-adjoint endomorphisms on finite-dimensional inner product spaces.

* Functional analysis
  - [PR #10145](https://github.com/leanprover-community/mathlib/pull/10145) Introduce `C*`-algebras
  - [PR #10530](https://github.com/leanprover-community/mathlib/pull/10530) Prove properties of spectrum in a 
  Banach algebra
```

> * Convergence.

Reviewer 6: I don't think this requires a special section (especially after moving the linear algebra PRs from the analysis section).

  - [PR #10073](https://github.com/leanprover-community/mathlib/pull/10073) Convergence of a sequence which does not oscillate infinitely

>   - [PR #10258](https://github.com/leanprover-community/mathlib/pull/10258) Prove that, if u_n is subadditive, then u_n / n converges.

Reviewer 6: This can be TeXed.

* Seminorms
  - [PR #9097](https://github.com/leanprover-community/mathlib/pull/9097) Define the Minkowski functional

>   - [PR #10203](https://github.com/leanprover-community/mathlib/pull/10203) A seminorm is convex

Reviewer 2: Not sure this one is worth a mention. Maybe we can put #9097 in a Convexity category along with maybe #10123 and some of [Oliver's PR](https://github.com/leanprover-community/mathlib/commits?author=ocfnash&before=42d3cbc350aea035e434da5652189e441c245bc2+35) about `barycentric_coords` and `continuous_affine_map`?

* Homotopy theory.

>   - [PR #9865](https://github.com/leanprover-community/mathlib/pull/9865) Add `homotopic` for `continuous_map`s.

Reviewer 6: I think this can be removed to keep only the next one.

>   - [PR #10195](https://github.com/leanprover-community/mathlib/pull/10195) The functor from `Top` to `Groupoid`

Reviewer 6: This is a very bad title. It should be: The fundamental groupoid, as a functor from...

* Sheafification.
  This series of PRs constructs the sheafification of presheaves with respect to a Grothendieck topology on a category.
  It is an important building block in setting up algebraic geometry,
  as well as a requirement for the Liquid Tensor Experiment (LTE).
  - [PR #10284](https://github.com/leanprover-community/mathlib/pull/10284) `P⁺` for a presheaf `P`.
  - [PR #10297](https://github.com/leanprover-community/mathlib/pull/10297) If `P` is a sheaf, then the map from `P` to `P⁺` is an isomorphism.
  - [PR #10303](https://github.com/leanprover-community/mathlib/pull/10303) The sheafification of a presheaf.
  - [PR #10328](https://github.com/leanprover-community/mathlib/pull/10328) `Sheaf_to_presheaf` creates limits
  - [PR #10334](https://github.com/leanprover-community/mathlib/pull/10334) `Sheaf J D` has colimits.
  - [PR #10401](https://github.com/leanprover-community/mathlib/pull/10401) `SheafedSpace` has colimits
  A corollary is that the category of sheaves
  with values in a suitable concrete abelian category is itself an abelian category.
  This has been formalised for LTE, and should appear in mathlib in the near future.

* Other.

>   - [PR #10188](https://github.com/leanprover-community/mathlib/pull/10188) Add definition and lemmas about open thickenings of subsets

Reviewer 6: I think this is worth mentioning, especially as it's explicitly preparation for something that will definitely be mentioned.

  Those are ingredients for the LYM inequality, Sperner's theorem and the Kruskal-Katona theorem.

* Measure theory.
  - [PR #9920](https://github.com/leanprover-community/mathlib/pull/9920) Define uniform pmf on an inhabited fintype
  - [PR #10070](https://github.com/leanprover-community/mathlib/pull/10070) More on Radon-Nikodym derivatives
  - [PR #10072](https://github.com/leanprover-community/mathlib/pull/10072) An ae measurability criterion for ennreal-valued functions
  - [PR #10057](https://github.com/leanprover-community/mathlib/pull/10057) Define Vitali families
  - [PR #10101](https://github.com/leanprover-community/mathlib/pull/10101) Differentiation of measures
This sequence of PRs shows the main theorem on differentiation of measures: given two measures `ρ` and `μ` on a finite-dimensional real vector space, the ratio `ρ (ball x r) / μ (ball x r)` converges `μ`-almost everywhere to the Radon-Nikodym derivative of `ρ` with respect to `μ` when `r` tends to `0`. The theorem is in fact proved in greater generality (Vitali families, as in Federer's famous book on geometric measure theory).

* Hilbert spaces

>   - [PR #9840](https://github.com/leanprover-community/mathlib/pull/9840) Rayleigh quotient
>   - [PR #9995](https://github.com/leanprover-community/mathlib/pull/9995) diagonalization theorem
>   - [PR #10317](https://github.com/leanprover-community/mathlib/pull/10317) more concrete diagonalization theorem
> This sequence of PRs establishes the connection between eigenvalues and the Rayleigh quotient, and proves the diagonalization theorem for self-adjoint endomorphisms on finite-dimensional inner product spaces.

Reviewer 3: ```suggestion
  - The sequence of PRs 
[PR #9840](https://github.com/leanprover-community/mathlib/pull/9840),
[PR #9995](https://github.com/leanprover-community/mathlib/pull/9995),
[PR #10317](https://github.com/leanprover-community/mathlib/pull/10317)
establishes the connection between eigenvalues and the Rayleigh quotient, and proves the diagonalization theorem for self-adjoint endomorphisms on finite-dimensional inner product spaces.
```
I see I misunderstood how the auto-formatting of blog posts happens.  Trying again ...

---
author: 'Mathlib community'
category: 'month-in-mathlib'

> date: 2021-12-01 06:58:04+00:00

Author: ```suggestion
date: 2021-12-06 06:58:04+00:00
```

      [PR #10328](https://github.com/leanprover-community/mathlib/pull/10328),
      [PR #10334](https://github.com/leanprover-community/mathlib/pull/10334),
      and [PR #10401](https://github.com/leanprover-community/mathlib/pull/10401)
      constructs the sheafification of presheaves with respect to
      a Grothendieck topology on a category, culminating with a construction
      of colimits in the category `SheafedSpace`. It is an important building block in
      setting up algebraic geometry, as well as a requirement for the
      [Liquid Tensor Experiment](https://github.com/leanprover-community/lean-liquid) 
      (LTE). A corollary is that the category of
      sheaves with values in a suitable concrete abelian category is itself
      an abelian category. This has been formalised for LTE, and should
      appear in mathlib in the near future.
    - Barycentric coordinates have been further studied in a series of PRs including for instance
      [PR #10320](https://github.com/leanprover-community/mathlib/pull/10320) relating them to determinants.      
    - [PR #10381](https://github.com/leanprover-community/mathlib/pull/10381)

>       proves the orthogonal group is generated by reflections

Reviewer 2: This is odd here. What about moving `#10381` to `Group theory` and `#10320` to `Linear algebra and functional analysis`?
Maybe this section can then be renamed to `Algebraic topology` alone.

Reviewer 5: I think this result fits well under "Geometry", so I agree with the current placement.
Also, naming this section "Algebraic Topology" alone (even after moving the orthogonal group PR) would not make sense at all, since a big chunk of this section is arguably about algebraic geometry.

Reviewer 2: As you wish. I maintain that barycentric coordinates do not belong here.

Reviewer 5: IMO this barycentric coordinates PR are fairly geometric. They are also fundamentally used in algebraic topology.

Reviewer 2: But all of linear algebra is fairly geometric 🤔

>     - [PR #10381](https://github.com/leanprover-community/mathlib/pull/10381)
>       proves the orthogonal group is generated by reflections

Reviewer 5: ```suggestion
    - [PR #10381](https://github.com/leanprover-community/mathlib/pull/10381)
      proves the orthogonal group is generated by reflections.
```

>     - [PR #10195](https://github.com/leanprover-community/mathlib/pull/10195)
>       defines the fundamental groupoid, as functor from `Top` to
>       `Groupoid`

Reviewer 5: ```suggestion
    - [PR #10195](https://github.com/leanprover-community/mathlib/pull/10195)
      defines the fundamental groupoid, as functor from `Top` to
      `Groupoid`.
```

      respectively defining antichains, shadow of a set family, UV compression.
      Those are ingredients for
      the [LYM inequality](https://en.wikipedia.org/wiki/Lubell%E2%80%93Yamamoto%E2%80%93Meshalkin_inequality),
      [Sperner's theorem](https://en.wikipedia.org/wiki/Sperner%27s_theorem)
      and the [Kruskal-Katona theorem](https://en.wikipedia.org/wiki/Sperner%27s_theorem).

* Linear algebra and functional analysis
    - The sequence of PRs 
      [PR #9840](https://github.com/leanprover-community/mathlib/pull/9840),
      [PR #9995](https://github.com/leanprover-community/mathlib/pull/9995),
      [PR #10317](https://github.com/leanprover-community/mathlib/pull/10317)
      establishes the connection between eigenvalues and the Rayleigh quotient,
      and proves the diagonalization theorem for self-adjoint endomorphisms on finite-dimensional inner product spaces.
    - [PR #10145](https://github.com/leanprover-community/mathlib/pull/10145) introduces $C^*$-algebras.
    - [PR #10530](https://github.com/leanprover-community/mathlib/pull/10530) proves properties of spectrum in a Banach algebra.

>     - [PR #9097](https://github.com/leanprover-community/mathlib/pull/9097) define the Minkowski gauge functional.

Author: ```suggestion
    - [PR #9097](https://github.com/leanprover-community/mathlib/pull/9097) defines the Minkowski gauge functional.
```

* Analysis
    - [PR #9791](https://github.com/leanprover-community/mathlib/pull/9791)

>       proves Picard-Lindelöf/Cauchy-Lipschitz theorem about ordinary differential equations.

Author: ```suggestion
      proves the Picard-Lindelöf/Cauchy-Lipschitz theorem about ordinary differential equations.
```

    - [PR #10057](https://github.com/leanprover-community/mathlib/pull/10057) defines Vitali families
      and [PR #10101](https://github.com/leanprover-community/mathlib/pull/10101) 
      shows the main theorem on differentiation of measures:
      given two measures `ρ` and `μ` on a finite-dimensional real vector space,
      the ratio `ρ (ball x r) / μ (ball x r)` converges `μ`-almost everywhere
      to the Radon-Nikodym derivative of `ρ` with respect to `μ` when `r` tends to `0`.
      The theorem is in fact proved in the more general context of Vitali families, as in Federer's famous book on geometric measure theory.

>     - [PR #10073](https://github.com/leanprover-community/mathlib/pull/10073) proves convergence of a sequence which does not oscillate infinitely

Reviewer 5: ```suggestion
    - [PR #10073](https://github.com/leanprover-community/mathlib/pull/10073) proves convergence of a sequence which does not oscillate infinitely.
```
